### ISSUE TITLE
[TW-1785] Purge command to remove deleted tasks

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1808]
_Paul Beckingham on 2009-05-25T21:42:34Z says:_ 

 New purge command to remove all old deleted tasks.  Perhaps configurable to remove only those over a certain age.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2009-05-25T21:42:34Z
Modified: 2016-03-20T18:45:34Z
```

**Comment by taskwarrior** [comment]
_David Patrick on 2009-05-26T11:49:47Z says:_ 

 purge command should move selected done tasks from "completed" and append to task-archive.data
I for one, will never want to fully delete all records of "done" tasks.
.. unless you're worried about disk space ? ;-)

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2009-05-26T17:58:00Z says:_ 

 Well, I have deleted tasks in my file from 2 1/2 years ago.  I want them gone.

**Comment by taskwarrior** [comment]
_David Patrick on 2009-05-27T09:02:37Z says:_ 

 Then you should be able to blithely delete (or purge archive) to permanently delete archive.data.

**Comment by taskwarrior** [comment]
_David Patrick on 2009-07-06T17:20:56Z says:_ 

 task purge [attrs] [#|date]
(note that [date] refers to the period from the beginning-of-time to [date])
to purge from files
completed.data
retired.data
archive.data

default action to move all [or as selected by # or date specifier] completed and/or retired tasks from completed.data to (append to) archive.data file. by default, IDs are not recycled.

this command ties into
PROPOSED NEW ID# HANDLING BEHAVIOR;
ID#s should remain static, and not be recycled until either "purge" command, or until the ID number is forced to add another digit.
That is to say that on rollover from 99 to 100, users should be prompted to purge and thereby recycle available ID#s, which are otherwise static. Same thing as ID#s threaten to roll over from 999 to 1000.
As task IDs roll over from 9999 to 10000, balloons and pidgins should be released, and a marching band should play outside your house. When IDs are recycled, the numbers change, but the order should not.

This way task IDs can remain static for a longer time.

**Comment by taskwarrior** [comment]
_David Patrick on 2009-07-15T23:52:48Z says:_ 

 suggested options (in development)

   purge [attrs] [#|date]           (so you could purge data from one proj:, for example)
   purge completed [attrs] [#|date] (don't recycle ID#s, or purge retired)
   purge ID [attrs] [#|date]        (recycle available IDs, don't purge completed or retired)
   purge retired [attrs] [#|date]   (not IDs or completed)
   purge archive [attrs] [#|date]   (to clear-out and delete forever)

**Comment by taskwarrior** [comment]
_Federico Hernandez on 2010-05-30T16:24:16Z says:_ 

 So you will be able to delete the now 3 1/2 year old done stuff. :-)

**Comment by taskwarrior** [comment]
_Tomas Babej on 2016-03-20T18:44:36Z says:_ 

 An version of this command, which accepts arbitrary filter was implemented in 2.6.0.

Paul can now delete the 9 1/2 year old done stuff :)
